src/compile/layer.ts

```diff
@@ -104,15 +104,43 @@ export class LayerModel extends Model {
     this.component.layout = parseLayerLayout(this);
   }
 
-  public parseScale() {
+  /**
+   * Convert the domain to an array of data refs. Also, thorw away sorting information
+   * since we always sort the domain when we union two domains.
+   */
+  private static normalizeDomain(domain: DataRefUnionedDomain | FieldRefUnionDomain | VgDataRef): VgDataRef[] {
```

Comment:
So they will be only used by layer, but not concat?  

Btw, FYI, we haven't used static method at all -- we mostly extracted then as a separate function.